Add Bluecore case study (#10727)
istio/istio.io: content/en/about/case-studies/bluecore/index.md

---
title: "Bluecore Leverages Istio to Migrate to Kubernetes"
linkTitle: "Bluecore Leverages Istio to Migrate to Kubernetes"
quote: "Istio enables an approach that allows you to start building right out of the box." 
author:
    name: "Shray Kumar"

>     image: "/img/authors/shray-kumar.jpg"

Reviewer: Please crop the face square to match the aesthetic on the other pages.

Author: @craigbox Both changes done!

companyName: "Bluecore"
companyURL: "https://bluecore.com/"
logo: "/logos/bluecore.png"
skip_toc: true
skip_byline: true
skip_pagenav: true
doc_type: article
sidebar_force: sidebar_case_study
type: case-studies

> weight: 30

Reviewer 2: @craigbox, do we have anything that talks about the weighting, or do we expect new case studies to be at the end? In this case the weight overlaps with `Bol.com`.

Reviewer: @pymander let's make it 51 please

Author: Done.

---

> Bluecore is a multichannel personalization platform that specializes in highly personalized emails, onsite messages, and paid media ads delivered at scale. Since Bluecore's unique offering requires processing customer recommendation data as well as sending a large quantity of personalized digital communications, they deal with ingesting, processing, and sending tons of data. it's a huge workload; during a recent Black Friday, they sent over 2 billion emails.

Reviewer: ```suggestion
Bluecore is a multichannel personalization platform that specializes in highly personalized emails, onsite messages, and paid media ads delivered at scale. Since Bluecore's unique offering requires processing customer recommendation data as well as sending a large quantity of personalized digital communications, they deal with ingesting, processing, and sending tons of data. It's a huge workload; during a recent Black Friday, they sent over 2 billion emails.
```

![2 billion emails and 12,000 Kubernetes pods](./bluecore-statistics.png)

> Along with volume, Bluecore needs to be speedy. They have service agreements with most of their partners guaranteeing that any customer will receive an email, onsite message, or ad within a few hours of a job being submitted, which means that processing speed is of vital importance. They accomplish this feat using both a monolithic application running on App Engine, and a Google Kubernetes Engine (GKE) cluster running about 12,000 pods.

Reviewer: ```suggestion
Along with volume, Bluecore needs to be speedy. They have service agreements with most of their partners guaranteeing that any customer will receive an email, onsite message, or ad within a few hours of a job being submitted, which means that processing speed is of vital importance. They accomplish this feat using both a monolithic application running on Google App Engine, and a Google Kubernetes Engine (GKE) cluster running about 12,000 pods.
```

## The Challenge: Monolithic architecture and increasing data traffic

Bluecore is a large operation with serious requirements around data transfer and processing. They were facing a daunting challenge; the amount of data they need to process continually increases. They knew their pipelines would get overloaded if their architecture couldn't be updated to handle increasing demands.

An examination of their architecture revealed that the monolithic application was going to be the biggest challenge. The Bluecore development team realized that to enable future growth, it was time to begin migrating to a more flexible and scalable infrastructure.

> Kubernetes offered a path to scale, and since Bluecore was already running on Google's App Engine, it seemed like migrating portions of their workflow to GKE would be a no-brainer. However, the majority of Bluecore's engineers didn't have sufficient experience with containerized applications. This could make migrating to GKE a painful transition.

Reviewer: ```suggestion
Kubernetes offered a path to scale, and since Bluecore was already running on App Engine, it seemed like migrating portions of their workflow to GKE would be a no-brainer. However, the majority of Bluecore's engineers didn't have sufficient experience with containerized applications. This could make migrating to GKE a painful transition.
```

Thankfully, Kumar and the infrastructure team were familiar with Istio. They worked closely with Al Delucca, principal engineer for Bluecore's data platform team, on a plan to begin implementing it.

"We had the problem," Delucca explains. "But whether or not Istio was the tool to solve it, that's what we had to figure out."

They discovered that Istio's feature set provided many solutions to existing challenges. Authorization was a big one. Incoming messages from partner applications needed to be authenticated. Istio could perform that authentication at the edge, which meant that each individual service didn't need to implement its own methods.

"We are able to push authentication and authorization to the edge, which takes the burden of understanding those systems away from our engineers," says Delucca. "It's not so much what they do with Istio directly, it's what Istio does for them without them knowing. That's the key—the big win for us."

> As their engineers began breaking the monolithic application down into services, they encountered another challenge that Istio was able to solve. Tracing calls to services looked as though it would be problematic. Many of the legacy functions didn't have clear documentation about their dependencies and requirements. This meant that performance issues and remote service calls could leave developers scratching their heads. Luckily, [Istio's distributed tracing](/docs/tasks/observability/distributed-tracing/) came to the rescue. With it, developers were able to pinpoint bottlenecks and services that needed bugfixes and additional work.

Reviewer: ```suggestion
As their engineers began breaking the monolithic application down into services, they encountered another challenge that Istio was able to solve. Tracing calls to services looked as though it would be problematic. Many of the legacy functions didn't have clear documentation about their dependencies and requirements. This meant that performance issues and remote service calls could leave developers scratching their heads. Luckily, [Istio's distributed tracing](/docs/tasks/observability/distributed-tracing/) came to the rescue. With it, developers were able to pinpoint bottlenecks and services that needed bug fixes and additional work.
```
